homeassistant/components/climate/broadlink.py

```diff
@@ -0,0 +1,414 @@
+                        config.get(CONF_MAX_TEMP))
+
+    add_devices([BroadlinkThermostat(hass, wt)])
+
+    def handle_set_schedule(service):
+        thermostat = WifiThermostat(config.get(CONF_MAC),
+                                    config.get(CONF_HOST),
+                                    config.get(CONF_NAME),
+                                    config.get(CONF_ADVANCED_CONFIG),
+                                    config.get(CONF_SCHEDULE_WEEKDAY),
+                                    config.get(CONF_SCHEDULE_WEEKDAY),
+                                    config.get(CONF_MIN_TEMP),
+                                    config.get(CONF_MAX_TEMP))
+        schedule_wd = service.data.get(CONF_SCHEDULE_WEEKDAY)
+        schedule_we = service.data.get(CONF_SCHEDULE_WEEKEND)
+        thermostat.set_schedule(
+            {CONF_WEEKDAY:
+             json.loads(schedule_wd.replace("'", '"'), cls=Decoder),
+             CONF_WEEKEND:
+             json.loads(schedule_we.replace("'", '"'), cls=Decoder)})
+
+    hass.services.register(DOMAIN, 'set_schedule',
+                           handle_set_schedule, schema=SET_SCHEDULE_SCHEMA)
+
+    def handle_set_advanced_conf(service):
+        thermostat = WifiThermostat(config.get(CONF_MAC),
+                                    config.get(CONF_HOST),
+                                    config.get(CONF_NAME),
+                                    config.get(CONF_ADVANCED_CONFIG),
+                                    config.get(CONF_SCHEDULE_WEEKDAY),
+                                    config.get(CONF_SCHEDULE_WEEKDAY),
+                                    config.get(CONF_MIN_TEMP),
+                                    config.get(CONF_MAX_TEMP))
+        advanced_conf = service.data.get(CONF_ADVANCED_CONFIG)
+        thermostat.set_advanced_config(
+            json.loads(advanced_conf.replace("'", '"'), cls=Decoder))
+
+        hass.services.register(DOMAIN, 'set_advanced_conf',
+                               handle_set_advanced_conf,
+                               schema=SET_ADVANCED_CONF_SCHEMA)
+
+        _LOGGER.debug("Wifi Thermostat: Component successfully added !")
+
+    return True
+
+
+class Decoder(json.JSONDecoder):
+    def decode(self, s):
+        result = super(Decoder, self).decode(s)
+        return self._decode(result)
+
+    def _decode(self, o):
+        if isinstance(o, str):
+            try:
+                return int(o)
+            except ValueError:
+                try:
+                    return float(o)
+                except ValueError:
+                    return o
+        elif isinstance(o, dict):
+            return {k: self._decode(v) for k, v in o.items()}
+        elif isinstance(o, list):
+            return [self._decode(v) for v in o]
+        else:
+            return o
+
+
+class WifiThermostat:
+    def __init__(self, mac, ip, name, advanced_config,
+                 schedule_wd, schedule_we, min_temp, max_temp):
+        self.HOST = ip
```

Comment:
Don't use capital letters for instance attributes. They should be lowercase snakecase.